### PR TITLE
[Feature] Add the support of BoxIouRotated op for ascend device

### PR DESCRIPTION
**Pull request by MrShadowY**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add the support of BoxIouRotated op for ascend device

## Modification

Add the support of BoxIouRotated op for ascend device

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.
Tests
image
image
image
image

### CODE REVIEW

**Review by ckirchhoff2021 — APPROVED** [review]
/LGTM

File: tests/test_ops/test_box_iou_rotated.py [hunk 1]
```diff
@@ -76,14 +80,27 @@ def test_box_iou_rotated(self, device):
         boxes2 = torch.from_numpy(np_boxes2).to(device)
 
         # test cw angle definition
+        # gpu
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:83]:
```suggestion
```

File: tests/test_ops/test_box_iou_rotated.py [hunk 2]
```diff
@@ -76,14 +80,27 @@ def test_box_iou_rotated(self, device):
...
+        boxes1_npu = torch.from_numpy(np_boxes1).npu()
+        boxes2_npu = torch.from_numpy(np_boxes2).npu()
+
+        ious = box_iou_rotated(boxes1_npu, boxes2_npu)
+        assert np.allclose(ious.cpu().numpy(), np_expect_ious, atol=1e-4)
+
+        ious = box_iou_rotated(boxes1_npu, boxes2_npu, aligned=True)
+        assert np.allclose(
+            ious.cpu().numpy(), np_expect_ious_aligned, atol=1e-4)
+
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:101]:
```suggestion
```

File: tests/test_ops/test_box_iou_rotated.py [hunk 3]
```diff
@@ -76,14 +80,27 @@ def test_box_iou_rotated(self, device):
...
         # test ccw angle definition
+        # gpu
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:103]:
```suggestion
```

File: tests/test_ops/test_box_iou_rotated.py [hunk 4]
```diff
@@ -93,6 +110,17 @@ def test_box_iou_rotated(self, device):
...
+        boxes1_npu[..., -1] *= -1
+        boxes2_npu[..., -1] *= -1
+        ious = box_iou_rotated(boxes1_npu, boxes2_npu, clockwise=False)
+        assert np.allclose(ious.cpu().numpy(), np_expect_ious, atol=1e-4)
+
+        ious = box_iou_rotated(
+            boxes1_npu, boxes2_npu, aligned=True, clockwise=False)
+        assert np.allclose(
+            ious.cpu().numpy(), np_expect_ious_aligned, atol=1e-4)
+
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:123]:
```suggestion
```

File: tests/test_ops/test_box_iou_rotated.py [hunk 5]
```diff
@@ -159,14 +191,28 @@ def test_box_iou_rotated_iof(self, device):
         boxes2 = torch.from_numpy(np_boxes2).to(device)
 
         # test cw angle definition
+        # gpu
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:194]:
```suggestion
```

File: tests/test_ops/test_box_iou_rotated.py [hunk 6]
```diff
@@ -159,14 +191,28 @@ def test_box_iou_rotated_iof(self, device):
...
+        boxes2_npu = torch.from_numpy(np_boxes2).npu()
+
+        ious = box_iou_rotated(boxes1_npu, boxes2_npu, mode='iof')
+        assert np.allclose(ious.cpu().numpy(), np_expect_ious, atol=1e-4)
+
+        ious = box_iou_rotated(
+            boxes1_npu, boxes2_npu, mode='iof', aligned=True)
+        assert np.allclose(
+            ious.cpu().numpy(), np_expect_ious_aligned, atol=1e-4)
+
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:213]:
```suggestion
```

File: tests/test_ops/test_box_iou_rotated.py [hunk 7]
```diff
@@ -159,14 +191,28 @@ def test_box_iou_rotated_iof(self, device):
...
         # test ccw angle definition
+        # gpu
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:215]:
```suggestion
```

File: tests/test_ops/test_box_iou_rotated.py [hunk 8]
```diff
@@ -176,3 +222,15 @@ def test_box_iou_rotated_iof(self, device):
...
+        boxes1_npu[..., -1] *= -1
+        boxes2_npu[..., -1] *= -1
+        ious = box_iou_rotated(
+            boxes1_npu, boxes2_npu, mode='iof', clockwise=False)
+        assert np.allclose(ious.cpu().numpy(), np_expect_ious, atol=1e-4)
+
+        ious = box_iou_rotated(
+            boxes1_npu, boxes2_npu, mode='iof', aligned=True, clockwise=False)
+        assert np.allclose(
+            ious.cpu().numpy(), np_expect_ious_aligned, atol=1e-4)
```
Comment (zhouzaida) [inline tests/test_ops/test_box_iou_rotated.py:236]:
```suggestion
```

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**